server/src/test/java/org/elasticsearch/common/util/MapsTests.java

```diff
@@ -123,6 +127,51 @@ public void testDeepEquals() {
         assertFalse(Maps.deepEquals(map, mapModified));
     }
 
+    public void testCollectToUnmodifiableSortedMap() {
+        SortedMap<String, String> canadianProvinces = Stream.of(new Tuple<>("ON", "Ontario"),
+                new Tuple<>("QC", "Quebec"),
+                new Tuple<>("NS", "Nova Scotia"),
+                new Tuple<>("NB", "New Brunswick"),
+                new Tuple<>("MB", "Manitoba"))
+            .collect(Maps.toUnmodifiableSortedMap(Tuple::v1, Tuple::v2));
+
+        assertThat(canadianProvinces, equalTo(new TreeMap<>(Maps.ofEntries(
+            Stream.of(new AbstractMap.SimpleEntry<>("ON", "Ontario"),
+                    new AbstractMap.SimpleEntry<>("QC", "Quebec"),
+                    new AbstractMap.SimpleEntry<>("NS", "Nova Scotia"),
+                    new AbstractMap.SimpleEntry<>("NB", "New Brunswick"),
+                    new AbstractMap.SimpleEntry<>("MB", "Manitoba"))
+                .collect(Collectors.toList())
+        ))));
+        expectThrows(UnsupportedOperationException.class, () -> canadianProvinces.put("BC", "British Columbia"));
+    }
+
+    public void testCollectRandomListToUnmodifiableSortedMap() {
+        List<Tuple<String, String>> tuples = randomList(1, 100,
+            () -> new Tuple<>(randomAlphaOfLength(10), randomAlphaOfLength(10)));
+
+        SortedMap<String, String> sortedTuplesMap = tuples.stream().collect(Maps.toUnmodifiableSortedMap(Tuple::v1, Tuple::v2));
+
+        String previous = "";
+        for (Map.Entry<String, String> entry : sortedTuplesMap.entrySet()) {
+            assertThat(entry.getKey().compareTo(previous) >= 0, equalTo(true));
+            previous = entry.getKey();
+        }
+    }
```

Comment:
Can we also assert that `sortedTyplesMap.keySet().equals(tuples.stream().map(Tuple::v1).collect(Collectors.toSet())`?

And also a validation that the resulting value in the map for each key is correct?

Reply:
I think you missed the second part of this comment?

Reply:
I believe I added this check on the line 160 https://github.com/elastic/elasticsearch/pull/77116/files#diff-e0297f09dc66a7f485aa41cccde6a1c3ae109bb903aa6d6af0267c00741d297eR160. Can move it down, though.

Reply:
That line verifies that the keys match. The second part of my comment was:

> And also a validation that the resulting value in the map for each key is correct?

This concerns the values in the map, I would like to see us validate that the collected map has the right values for every key.

Reply:
I would like to see the values verified too (see above).

Reply:
Done!